src/Highway.Data.EntityFramework/Factories/DomainRepositoryFactory.cs

```diff
@@ -45,75 +42,45 @@ public IRepository Create<T>()
+        /// <typeparam name="T">Domain for repository</typeparam>
+        /// <returns>Domain-specific readonly repository</returns>
+        public IReadonlyRepository CreateReadonly<T>()
+            where T : class, IDomain
         {
-        }
+            var domain = _domains.OfType<T>().SingleOrDefault();
+            var context = new ReadonlyDomainContext<T>(domain);
 
-        /// <summary>
-        ///     Creates a repository factory for the supplied list of domains
-        /// </summary>
-        public RepositoryFactory(string connectionString, IMappingConfiguration mappings, IContextConfiguration contextConfiguration)
-            : this(connectionString, mappings, contextConfiguration, new NoOpLogger())
-        {
+            return new ReadonlyDomainRepository<T>(context, domain);
         }
 
         /// <summary>
-        ///     Creates a repository factory for the supplied list of domains
+        ///     Creates a readonly repository for the requested domain
         /// </summary>
-        public RepositoryFactory(string connectionString, IMappingConfiguration mappings, ILog logger)
-            : this(connectionString, mappings, new DefaultContextConfiguration(), logger)
+        /// <param name="type">Domain for repository</param>
+        /// <returns>Domain-specific readonly repository</returns>
+        public IReadonlyRepository CreateReadonly<T>(Type type)
         {
+            return (IReadonlyRepository)CreateRepository(type, typeof(ReadonlyDomainContext<>), typeof(ReadonlyDomainRepository<>));
         }
 
-        /// <summary>
-        ///     Creates a repository factory for the supplied list of domains
-        /// </summary>
-        public RepositoryFactory(string connectionString, IMappingConfiguration mappings, IContextConfiguration contextConfig, ILog logger)
+        private object CreateRepository(Type domainType, Type contextType, Type repositoryType)
         {
-            _connectionString = connectionString;
-            _mappings = mappings;
-            _contextConfig = contextConfig;
-            _logger = logger;
-        }
+            Type[] typeArgs = { domainType };
 
-        /// <summary>
-        ///     Creates a repository for the requested domain
-        /// </summary>
-        /// <returns>Domain specific repository</returns>
-        public IRepository Create()
-        {
-            return new Repository(new DataContext(_connectionString, _mappings, _contextConfig, _logger));
+            var domain = _domains.SingleOrDefault(x => x.GetType() == domainType);
+            var contextCtor = contextType.MakeGenericType(typeArgs);
+            var untypedObject = Activator.CreateInstance(contextCtor, domain);
+
+            var repositoryCtor = repositoryType.MakeGenericType(typeArgs);
+            var repo = Activator.CreateInstance(repositoryCtor, untypedObject, domain);
+
+            return (IRepository)repo;
         }
     }
 }
```

Comment:
This file winds up looking cluttered in GitHub's diff.  Just view the new file.  The change is pretty simple:

- Added new methods to implement IReadonlyDomainRepositoryFactory (no need to two classes in the case)
- Split RepositoryFactory into its own file.
- Extract a private _CreateRepository_ method to back the non-generic create methods.